samples/drivers/led_max7219/src/main.c

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2015 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <zephyr.h>
+#include <misc/printk.h>
+#include <device.h>
+#include <gpio.h>
+#include <spi.h>
+
+#define SPI_DEV  DT_SPI_1_NAME
+#define GPIO_DEV DT_GPIO_P0_DEV_NAME
+
+/* Any GPIO pin that suits your design */
+#define PIN_CS   18
+
+struct device *spi_dev;
+struct device *gpio_dev;
+
+struct spi_cs_control cs;
+
+void spi_init(void)
+{
+	spi_dev = device_get_binding(SPI_DEV);
+	if (!spi_dev) {
+		printk("SPI: Device driver not found.\n");
+	} else {
+		printk("SPI device found\n");
+	}
+}
+
+struct spi_cs_control cs = {
+	.gpio_pin = PIN_CS,
+	.delay = 0,
+};
+
+struct spi_config spi_cfg = {
+	.frequency = 1000000U,  /* 1 MHz, but MAX7219 supports up to 10 MHz */
+	.operation = SPI_TRANSFER_MSB | SPI_WORD_SET(8) | SPI_CS_ACTIVE_HIGH,
+	.slave = 0,
+	.cs = &cs,
+};
+
+#define SPI_WORDS 2
+u8_t buffer_tx[SPI_WORDS];
+
+const struct spi_buf tx_buf = {
+	.buf = buffer_tx,
+	.len = SPI_WORDS,
+};
+
+const struct spi_buf_set tx = {
```

Comment:
`buffer_tx`, `tx_buf`, and `tx` are used only inside `max_send_byte`. Why make them global then?

Reply:
Moved to max_send_command.